relayer-cli/src/commands/tx/client.rs

```diff
@@ -163,3 +164,64 @@ impl Runnable for TxUpgradeClientCmd {
         }
     }
 }
+
+#[derive(Clone, Command, Debug, Options)]
+pub struct TxUpgradeClientsCmd {
+    #[options(free, required, help = "identifier of the chain that must be upgraded")]
+    src_chain_id: ChainId,
+}
+
+impl Runnable for TxUpgradeClientsCmd {
+    fn run(&self) {
+        let config = app_config();
+
+        let outputs: Vec<Output> = config
+            .chains
+            .iter()
+            .map(|chain| match self.upgrade_chain(&config, &chain.id) {
+                Ok(receipts) => Output::success(receipts),
+                Err(e) => Output::error(e),
+            })
+            .collect();
+
+        Output::combined(outputs.into_iter()).exit()
+    }
+}
+
+impl TxUpgradeClientsCmd {
+    fn upgrade_chain(
+        &self,
+        config: &config::Reader<CliApp>,
+        dst_chain_id: &ChainId,
+    ) -> Result<Vec<IbcEvent>, Error> {
+        let chains = ChainHandlePair::spawn(&config, &self.src_chain_id, dst_chain_id)?;
+
+        let req = QueryClientStatesRequest {
+            pagination: ibc_proto::cosmos::base::query::pagination::all(),
+        };
+        let client_ids: Vec<ClientId> = chains
+            .dst
+            .query_clients(req)
+            .map_err(|e| Kind::Query.context(e))?
+            .iter()
+            .filter_map(|cs| {
+                if self.src_chain_id == cs.client_state.chain_id() {
+                    Some(cs.client_id.clone())
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        debug_assert!(client_ids.len() < 2);
```

Comment:
I'm curious what is this assert for?

Reply:
I wrongly presumed there could be only one client per host-target pair.  😓 Fixed!